### ISSUE TITLE
support endpoint default configurations

### ISSUE THREAD

**Issue by OwenPriceSkelly** [Garden-AI/groundhog#34]
I was surprised that anvil required `scheduler_options="#SBATCH --gpus-per-node=1"` to be set in order to run on gpu. might be nice if there was a `.groundhog.toml` or something where you could set per-endpoint defaults

**Comment by OwenPriceSkelly** [comment]
claude brainstorming dump: 
## Support PEP 723 `[tool.groundhog]` Configuration in Script Metadata

### Summary

Enable script-level configuration through PEP 723 inline script metadata `[tool.groundhog]` sections. This makes groundhog scripts self-contained and portable, supporting the primary use case of scientists sharing standalone scripts with one another.

### Motivation

Currently, users must hardcode endpoint configuration in every `@hog.function()` decorator:

```python
@hog.function(endpoint="5aafb4c1-27b2-40d8-a038-a0277611868f", account="cis250461")
def compute(x: int) -> int:
    return x * 2
```

This creates several problems:

1. **Not portable** - Scripts contain user-specific accounts and endpoint IDs
2. **Repetitive** - Same config repeated across multiple functions in a script
3. **Version control issues** - Personal accounts/secrets get committed
4. **Machine-specific gotchas hidden** - Different HPC systems have different scheduler requirements (e.g., Anvil GPU queues require `gpus_per_node`), but there's no place to encode defaults

### Proposed Solution

Support reading configuration from `[tool.groundhog]` in PEP 723 inline script metadata (`# /// script` blocks). This is groundhog's primary "unit of sharing" - standalone Python scripts.

### Detailed Design

#### PEP 723 Inline Script Metadata

```python
# /// script
# requires-python = ">=3.10"
# dependencies = ["numpy"]
#
# [tool.groundhog]
# endpoint = "anvil"  # or UUID
# account = "my-allocation-123"
# queue = "shared"
# ///

import groundhog_hpc as hog

@hog.function()  # Inherits endpoint + account + queue from metadata
def compute(x: int) -> int:
    import numpy as np
    return int(np.sum(range(x)))

@hog.function(queue="gpu-debug", gpus_per_node=1)  # Override queue, inherit rest
def gpu_compute(x: int) -> int:
    import torch
    return x * 2

@hog.harness()
def main():
    result = compute.remote(100)
    print(result)
```

#### Schema for `[tool.groundhog]`

Any fields that are valid in `user_endpoint_config` can appear in `[tool.groundhog]`:

```toml
[tool.groundhog]
# Required
endpoint = "anvil"  # Friendly name or UUID

# Common fields (all optional)
account = "my-allocation-123"
queue = "shared"
walltime = "01:00:00"
nodes = 1
gpus_per_node = 1  # When using GPU queues
# ... any other user_endpoint_config fields
```

**Optional: Named Profiles**

For scripts with multiple distinct workload types:

```python
# /// script
# [tool.groundhog.profiles.cpu]
# endpoint = "anvil"
# account = "my-allocation"
# queue = "shared"
#
# [tool.groundhog.profiles.gpu]
# endpoint = "anvil"
# account = "my-allocation"
# queue = "gpu-debug"
# gpus_per_node = 1
# ///

@hog.function(profile="cpu")
def cpu_task(): ...

@hog.function(profile="gpu")
def gpu_task(): ...
```

#### Configuration Resolution Order

```python
config = {}

# 1. Built-in endpoint profile (if endpoint known - future work)
if endpoint in BUILTIN_ENDPOINTS:
    config.update(BUILTIN_ENDPOINTS[endpoint].defaults)

# 2. User config file (~/.config/groundhog/config.toml - future work)
config.update(load_user_config().get(endpoint, {}))

# 3. Script [tool.groundhog] metadata (THIS ISSUE)
if script_has_pep723_metadata():
    config.update(parse_pep723_tool_groundhog())

# 4. Named profile (if specified)
if profile := function_kwargs.get('profile'):
    config.update(resolve_profile(profile))

# 5. Function decorator kwargs
config.update(function_kwargs)

# 6. Call-time .remote(user_endpoint_config={...})
config.update(call_time_overrides)
```

### Implementation Notes

1. **PEP 723 parsing**: Extend existing PEP 723 parsing (already done for dependencies) to extract `[tool.groundhog]`
2. **Validation**: Validate schema at load time, provide helpful errors for unknown fields
3. **Endpoint name resolution**: Support both friendly names (`"anvil"`) and UUIDs in `endpoint` field
4. **Backward compatibility**: All existing scripts continue to work unchanged
5. **Scope**: Only PEP 723 inline metadata in standalone scripts, not `pyproject.toml` files

### Benefits

1. **Portable scripts** - Share `.py` file with PEP 723 metadata, recipient just updates `account` field
2. **DRY** - Configure once per script instead of per function
3. **Self-documenting** - Script metadata shows what it needs to run
4. **Reproducible** - Config versioned with code alongside dependencies
5. **Gradual adoption** - Existing scripts work unchanged, opt-in enhancement
6. **Consistent with dependencies** - Same mechanism as PEP 723 dependency specification

### Example: Before/After

**Before:**
```python
# Must edit every function decorator to share script
@hog.function(endpoint="5aafb4c1-...", account="cis250461", queue="shared")
def task1(): ...

@hog.function(endpoint="5aafb4c1-...", account="cis250461", queue="shared")
def task2(): ...

@hog.function(endpoint="5aafb4c1-...", account="cis250461", queue="gpu-debug", gpus_per_node=1)
def task3(): ...
```

**After:**
```python
# /// script
# requires-python = ">=3.10"
# dependencies = ["numpy"]
#
# [tool.groundhog]
# endpoint = "anvil"
# account = "cis250461"
# queue = "shared"
# ///

@hog.function()  # Inherits from metadata
def task1(): ...

@hog.function()  # Inherits from metadata
def task2(): ...

@hog.function(queue="gpu-debug", gpus_per_node=1)  # Override queue
def task3(): ...
```

Recipient updates `account = "their-account"` in one place, everything works.

### Future Work (Not This Issue)

- Built-in endpoint profiles shipped with groundhog
- User config file `~/.config/groundhog/config.toml`
- CLI helpers: `hog config set`, `hog config show`
- Endpoint validation rules (e.g., "gpu queue requires gpus_per_node")
- Support for `pyproject.toml` in package projects (if needed)

### Related

- Dependency specification via PEP 723 already implemented
- `user_endpoint_config` dict already supports all needed fields
- `GROUNDHOG_SCRIPT_PATH` env var already tracks script location
